exonum/src/blockchain/schema.rs

```diff
@@ -373,42 +388,55 @@ impl<'a> Schema<&'a mut Fork> {
     /// [1]: struct.Schema.html#method.block_txs
     pub fn block_txs_mut(&mut self, height: Height) -> ProofListIndex<&mut Fork, Hash> {
         let height: u64 = height.into();
-        ProofListIndex::with_prefix(BLOCK_TXS, gen_prefix(&height), &mut self.view)
+        ProofListIndex::with_prefix(BLOCK_TXS, gen_prefix(&height), self.view)
     }
 
     /// Mutable reference to the [`precommits`][1] index.
     ///
     /// [1]: struct.Schema.html#method.precommits
     pub fn precommits_mut(&mut self, hash: &Hash) -> ListIndex<&mut Fork, Precommit> {
-        ListIndex::with_prefix(PRECOMMITS, gen_prefix(hash), &mut self.view)
+        ListIndex::with_prefix(PRECOMMITS, gen_prefix(hash), self.view)
     }
 
     /// Mutable reference to the [`configs`][1] index.
     ///
     /// [1]: struct.Schema.html#method.configs
     pub fn configs_mut(&mut self) -> ProofMapIndex<&mut Fork, Hash, StoredConfiguration> {
-        ProofMapIndex::new(CONFIGS, &mut self.view)
+        ProofMapIndex::new(CONFIGS, self.view)
     }
 
     /// Mutable reference to the [`configs_actual_from`][1] index.
     ///
     /// [1]: struct.Schema.html#method.configs_actual_from
     pub fn configs_actual_from_mut(&mut self) -> ListIndex<&mut Fork, ConfigReference> {
-        ListIndex::new(CONFIGS_ACTUAL_FROM, &mut self.view)
+        ListIndex::new(CONFIGS_ACTUAL_FROM, self.view)
     }
 
     /// Mutable reference to the [`state_hash_aggregator`][1] index.
     ///
     /// [1]: struct.Schema.html#method.state_hash_aggregator
     pub fn state_hash_aggregator_mut(&mut self) -> ProofMapIndex<&mut Fork, Hash, Hash> {
-        ProofMapIndex::new(STATE_HASH_AGGREGATOR, &mut self.view)
+        ProofMapIndex::new(STATE_HASH_AGGREGATOR, self.view)
     }
 
     /// Mutable reference to the [`peers_cache`][1] index.
     ///
     /// [1]: struct.Schema.html#method.peers_cache
     pub fn peers_cache_mut(&mut self) -> MapIndex<&mut Fork, PublicKey, Connect> {
-        MapIndex::new("core.peers_cache", &mut self.view)
+        MapIndex::new("core.peers_cache", self.view)
+    }
+
+    /// Mutable reference to the [`consensus_messages_cache`][1] index.
+    ///
+    /// [1]: struct.Schema.html#method.consensus_messages
+    pub fn consensus_messages_cache_mut(&mut self) -> ListIndex<&mut Fork, RawMessage> {
```

Comment:
I'd say *all* mutating methods of the schema should have `pub(crate)` or lesser visibility, especially those dealing with indices directly. In this particular case, I would argue even read-only indices should be `pub(crate)` (OTOH, it may adversely impact testing, but in this case I'd say conditional compilation may still be preferable to exposing a potentially very dangerous interface).